### PR TITLE
Add a walkthrough document and linked on the ReadMe

### PR DESCRIPTION
**Pull request by julieheard**
Hi, I have made a walkthough for how to use the ATH when testing changes made to a local version of Jenkins. I was not entirely confident about where to link it in the main ReadMe, please let me know if you think it is not in the correct place.

Hopefully someone will find this useful 😄 

### Testing done
This is a documentation PR, I have tested locally by veiwing the files and making sure they display corectly.

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: docs/WALKTHROUGH.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+In this walkthrough I am going to show you how to locally use this ATH to test changes made in a local copy of Jenkins.
+
+## Getting Started
+
+In this walkthrough I will be using Docker, IntelliJ and a terminal.
+
+1) Make your own fork of https://github.com/jenkinsci/jenkins
+2) Clone the fork to your local machine
+> **Note:** The next two steps are optional, this makes it easy to make a PR to Jenkins if you wish to do so later
```
Comment (jtnord) [inline docs/WALKTHROUGH.md:11]:
```suggestion
> **Note** The next two steps are optional, this makes it easy to make a PR to Jenkins if you wish to do so later
```

I think without the colon. https://github.com/orgs/community/discussions/16925

File: docs/WALKTHROUGH.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+3) Create a new branch and give it a sensible name
+4) Make your changes to your copy of Jenkins, commit changes to your branch, feel free to push your changes to your local fork 
+
+
+
+## Running the ATH
+
+When you have made changes to your local copy of Jenkins, in a terminal window `cd` into your local copy of Jenkins and run `mvn verify`, this will run the integration tests inside the jenkins repo. 
+
+When you are ready to test your changes with this ATH, you will need to run `mvn install` on your local copy of Jenkins. This achieves multiple things:
```
Comment (jtnord) [inline docs/WALKTHROUGH.md:21]: Just `mvn package` should suffice

File: docs/WALKTHROUGH.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    [INFO] BUILD SUCCESS
+    [INFO] ------------------------------------------------------------------------
+    ```
+
+2) Navigate in your local Jenkins repo to `…\war\target` and you will see a war file for you to use later
+
+### Testing with this ATH
+1) Clone a copy of this ATH repo to your local machine and `cd` into the directory
+2) Open the `pom.xml` in your local copy of the ATH
+3) Change the `<jenkins.version>` to your SNAPSHOT version of jenkins (mine is `2.415-SNAPSHOT`)
```
Comment (jtnord) [inline docs/WALKTHROUGH.md:50]: This is not what you want to do.

Make a link (mklink)  from `jenkins.war` to ../Jenkins/war/target/Jenkins.war
Or copy the war (where ../Jenkins is the path to the Jenkins repo you cloned)

File: docs/WALKTHROUGH.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+> **Note:** There are many ways to run these tests, I will be running the tests using Windows, your commands may differ slightly if using a different OS. See the [README.md](README.md) for alternatives and configuration options.  
+
+4) In a terminal window type `set JENKINS_WAR=C:\Users\julie\jenkins\war\target\jenkins.war` This sets a new environment variable called `JENKINS_WAR` to the path of your copy of the jenkins war *(You created this in the `mvn install step`, change it to your directory path instead of mine)*
```
Comment (jtnord) [inline docs/WALKTHROUGH.md:54]: Ok so another way to set, so you definitely do not need to set the Jenkins
Version earlier!

File: docs/WALKTHROUGH.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+6) Copy the docker command that was output to your terminal after the last step, and paste it into the same terminal window. This will run the docker container with your local copy of Jenkins.
+
+**You are now ready to run tests.**
+
+If you want to run all the tests, run `mvn test`, if you want to run a specific test run `mvn test -Dtest=ArtifactsTest` *(change the test name to the one you wish to run)* tests can be found in the src/test/java directory of this ATH repository.
+
+### Tips and tricks
+- When running your tests with `mvn test`, adding the parameter `-DforkCount=` with a number higher than 1 will run tests in parallel. This is useful for running tests in a shorter amount of time. See [here](https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html) for more details.
+- If you make changes to your local jenkins version after running ATH tests, and you want to re-run the tests again, make sure you:
+  - run `mvn install` again in your local jenkins repository and double-check your SNAPSHOT version
```
Comment (jtnord) [inline docs/WALKTHROUGH.md:65]: Just package if you do not mess with the ath pom

File: docs/WALKTHROUGH.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  - stop the docker container you were running previously and re-run it. You can re-run `vars.cmd` again to get the docker command.
```
Comment (jtnord) [inline docs/WALKTHROUGH.md:66]: You shouldn't need to stop and restart the docker container.

**Comment by julieheard** [comment]
All changes made, thank you for the feedback 🙂

**Review by jtnord — APPROVED** [review]
Thanks @julieheard

**Pull request merged**